### ISSUE TITLE
Recover type information

### ISSUE THREAD

**Issue by frabert** [lifting-bits/rellic#190]
Currently, some type information (e.g. `const`, `volatile`) regarding variables is lost.
For example, this source code

```c
struct foo {
  int x;
};

typedef struct foo foo_t;

int main(void) {
  const volatile foo_t **a = 0;
  void *b = a;
}
```

roundtrips to:

```c
unsigned int main();
struct struct_foo {
    unsigned int x;
};
unsigned int main() {
    struct struct_foo **a; // lost const volatile
    unsigned char *b; // was void* before
    a = (void *)0U;
    b = (unsigned char *)a;
    return 0U;
}
```

It would be nice to recover the original type information when possible.

**Comment by pgoodman** [comment]
Do you see `"omnipotent char"` show up anywhere in debug info? We might be able to see that and use `char`. There is [`CharIsSigned`](https://code.woboq.org/llvm/clang/include/clang/Basic/LangOptions.def.html#clang::LangOptionsBase::CharIsSigned), and [here](https://code.woboq.org/llvm/lldb/source/Plugins/TypeSystem/Clang/TypeSystemClang.cpp.html#530) is how LLDB infers this.

**Comment by frabert** [comment]
Yes, `"omnipotent char"` does show up as part of TBAA attributes on `load` and `store` instructions.
I've tried searching for something but couldn't really grasp what that means.

**Comment by pgoodman** [comment]
Perhaps it's actually a sort of joke type for pointers. Perhaps if we know if it's meant to be a `char`, but want to hedge against signed/unsigned, we could have a typedef like `typedef signed char char_t` or `typedef unsigned char char_t`, or have a `char_t` and a `uchar_t`.
